### ISSUE TITLE
Rule proposal: `consistent-empty-array-spread`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2074]
### Description

When spreading a ternary in array, we can use both

```js
[
	element1,
	...(a ? [b, c] : []),
]
```

and

```js
[
	element1,
	...(a ? [b, c] : ''),
]
```

but it's bad to have different style in one project, should enforce consistent style.

Personally, I slightly prefer to use empty array when the consequent is array, but empty string should be cheaper. I hope this rule default to enforce use the same type for "consequent" and "alternate".

### Fail

```js
[
	element1,
	...(a ? [b, c] : ''), // use `[]` since "consequent" is array
]
```

```js
[
	element1,
	...(a ? 'bc' : []), // should use `""` since "consequent" is string
]
```

### Pass

```js
[
	element1,
	...(a ? [b, c] : []), // use `[]` since "consequent" is array
]
```

```js
[
	element1,
	...(a ? 'bc' : ''), // should use `""` since "consequent" is string
]
```

**Comment by silverwind** [comment]
Would name the rule specifical for array, because object spread has different rules and also accepts things like `{...null}` or `{...false}` which array spread doesn't where such values are `TypeError`s.

**Comment by sindresorhus** [comment]
Accepted
